### ISSUE TITLE
Likelihood ratio test? 

### ISSUE THREAD

**Issue by lucia-ramirez** [GabrielHoffman/variancePartition#50]
Is it possible to do a likelihood ratio test like in DESeq2? I know that is possible to test multiple coefficients with dream, but that would mean to compare against a baseline. For example, let's say I have different time points (1,2,3,4) and I want to look for interactions between sex and the time points.  If I do this: 

~ 0 + time*sex  + (1|Donor) 
  
And then test the interaction terms, in theory the results would be against the time point 1, when I am interested in the overall effect. 

Thanks,

Lucia

**Comment by GabrielHoffman** [comment]
Hi Lucia, 
While `dream` does not support LRT's, you should be able to get anything you want from a Wald test.  Here is some code to test a sex by time point interaction effect on gene expression: 

```r
library(variancePartition)

data(varPartData)

# Generate sex variable
info$Sex = 'F' 
i = match(info$Individual, levels(info$Individual)[1:12])
info$Sex[!is.na(i)] = "M"

# Generate Time variable
info$Time = as.numeric(info$Batch)

# interaction formula
form <- ~ Sex*Time + (1|Individual)

# fit model
fit = dream( geneExpr[1:10,], form, info)
#> Dividing work into 1 chunks...
#> 
#> Total:0.7 s
fit = eBayes(fit)

# extract coefs
colnames(coef(fit))
#> [1] "(Intercept)" "SexM"        "Time"        "SexM:Time"

topTable(fit, coef="SexM:Time")
#>              logFC     AveExpr          t    P.Value adj.P.Val         B
#> gene7  -0.76775901  -4.3799381 -2.2659476 0.02599073 0.1738071 -3.409562
#> gene9  -1.17413101  -2.3079042 -2.1439690 0.03476142 0.1738071 -3.599441
#> gene2  -0.43614619  -1.1281610 -1.3486775 0.18102770 0.6034257 -4.612469
#> gene6  -0.40710094  -3.1554790 -0.9268406 0.35659809 0.7025581 -4.974272
#> gene1  -0.23725677 -10.4664549 -0.8801524 0.38129993 0.7025581 -5.000908
#> gene3   0.19996284   0.1702122  0.7475813 0.45680551 0.7025581 -5.082549
#> gene10  0.22772650  -2.3673775  0.6904103 0.49179067 0.7025581 -5.119477
#> gene5   0.11650225   4.7187640  0.2655369 0.79122432 0.9738832 -5.276056
#> gene8  -0.04378256   0.9171386 -0.1232343 0.90221336 0.9738832 -5.291415
#> gene4   0.01006904  -4.5359748  0.0328340 0.97388318 0.9738832 -5.299613
#>              z.std
#> gene7  -2.22635023
#> gene9  -2.11112679
#> gene2  -1.33759744
#> gene6  -0.92186675
#> gene1  -0.87550364
#> gene3   0.74411731
#> gene10  0.68746354
#> gene5   0.26472110
#> gene8  -0.12286581
#> gene4   0.03273843
```

**Comment by lucia-ramirez** [comment]
Thanks a lot! I was initially treating time as a categorical variable because some of my time points are really far apart and may not have a linear relationship. 

For instance, in the example you put, would it be possible to test the interaction between batch and sex? 

Thanks

**Comment by lucia-ramirez** [comment]
I am actually getting an error when trying to do the contrasts matrix with your same example

```
library(variancePartition)

data(varPartData)

# Generate sex variable
info$Sex = 'F' 
i = match(info$Individual, levels(info$Individual)[1:12])
info$Sex[!is.na(i)] = "M"

# Generate Time variable
info$Time = as.numeric(info$Batch)
info$Time = factor(info$Time)

# interaction formula
form <- ~ Sex*Time + (1|Individual)

makeContrastsDream(form, info, contrasts = c("SexM:Time2 - SexM:Time3"))

Error in dimnames(x) <- dn : 
  length of 'dimnames' [1] not equal to array extent
In addition: Warning messages:
1: In SexM:Time2 :
  numerical expression has 8 elements: only the first used
2: In SexM:Time2 :
  numerical expression has 8 elements: only the first used
3: In SexM:Time3 :
  numerical expression has 8 elements: only the first used
4: In SexM:Time3 :
  numerical expression has 8 elements: only the first used
```

**Comment by GabrielHoffman** [comment]
I just fixed this bug in the new version: `1.25.12`

Does that work?

**Comment by lucia-ramirez** [comment]
Yes, it does. 

Thanks!
